src/graph.jl

```diff
@@ -0,0 +1,349 @@
+
+        v_props = props(input_graph, i)
+
+        set_indexing_prop!(G, i, :v_hash, v_props[:v_hash])
+
+        set_props!(G, i, 
+            Dict(:v_name=>v_props[:v_name], 
+                :v_type=>v_props[:v_type],
+                :used=>true))
+    end   
+    
+    
+    @info "The input graph contains $num_vertices_g unique vertices. \n The input edge list refers to $num_vertices_el unique vertices. \n The size of the intersection of these two sets is: $inter_cardinality."
+
+    set_indexing_prop!(G, :v_hash)
+
+    for e in eachrow(e_df)
+
+        edge_attrs = Dict(
+                        :e_rel=>e.edge_relation,
+                        :e_desc=>e.edge_description,
+                        :e_value=>e.value,
+                        :counter=>1)
+
+        # The src vertex is \notin G; we need to insert it before we can insert the edge
+        if !issubset([e.src_vhash], vertices_already_in_G)
+
+            src_v_hash = e.src_vhash
+                
+            src_v_attrs = Dict(
+                :v_name=>e.src_name,
+                :v_type=>e.src_vtype,
+                :used=>true)
+
+            set_indexing_prop!(G, counter, :v_hash, "$src_v_hash")
+            set_props!(G, counter, src_v_attrs)
+                
+            counter += 1
+            vertices_already_in_G = cat(dims=1, "$src_v_hash", vertices_already_in_G)
+            
+            @info "src vertex $src_v_hash was not in G, and has been inserted."
+        end
+        
+        # The dst vertex is \notin G; we need to insert it before we can insert the edge
+        if !issubset([e.dst_vhash], vertices_already_in_G)
+            
+            dst_v_hash = e.dst_vhash
+                
+            dst_v_attrs = Dict(
+                :v_name=>e.dst_name,
+                :v_type=>e.dst_vtype,
+                :used=>true)
+
+            set_indexing_prop!(G, counter, :v_hash, "$dst_v_hash")
+            set_props!(G, counter, dst_v_attrs)
+                
+            counter += 1
+            vertices_already_in_G = cat(dims=1, "$dst_v_hash", vertices_already_in_G)
+            @info "dst vertex $dst_v_hash was not in G, and has been inserted."
+        end
+        
+        # All vertices referenced are now \in G; we can insert the edge or increment an existing edge's counter
+        src_int_id = G[e.src_vhash, :v_hash]
+        dst_int_id = G[e.dst_vhash, :v_hash]
+            
+        # Add a new edge to the graph
+        if !has_edge(G, src_int_id, dst_int_id)
+
+
+            add_edge!(G, src_int_id, dst_int_id)
+            set_props!(G, Edge(src_int_id, dst_int_id), edge_attrs)
+
+        #  This edge already exists \in G; increment the (frequency) counter (currently used for weight; we will probably want to think about how to best normalize these counts)
+        else
+            cur_weight = get_prop(G, Edge(src_int_id, dst_int_id), :counter)
+            set_prop!(G, Edge(src_int_id, dst_int_id), :counter, cur_weight+1)
```

Comment:
Is this `:counter` related to the ` counter += 1` above?

Reply:
no, I'll fix that. am assuming though that integer weights is the way to go until we decide otherwise (?)

Reply:
yeah

Reply:
fixed; weight field is called `:weight` for now.